iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/LongRunInstructionCheck.java

```diff
@@ -0,0 +1,136 @@
+  public static final int DEFAULT_MIN_WORD_TO_TRIGGER = 7;
+  private static final String MESSAGE = "Split this RUN instruction line into multiple lines.";
+
+  @RuleProperty(
+    key = "maxLength",
+    description = "The maximum length that should not be exceeded by any RUN line instruction.",
+    defaultValue = "" + DEFAULT_MAX_LENGTH)
+  public int maxLength = DEFAULT_MAX_LENGTH;
+
+  @RuleProperty(
+    key = "minWordsToTrigger",
+    description = "The minimum amount of words required in a single line to trigger the check.",
+    defaultValue = "" + DEFAULT_MIN_WORD_TO_TRIGGER)
+  public int minWordsToTrigger = DEFAULT_MIN_WORD_TO_TRIGGER;
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(RunInstruction.class, this::checkInstruction);
+  }
+
+  private void checkInstruction(CheckContext ctx, RunInstruction runInstruction) {
+    List<TooLongLine> tooLongLines = computeTooLongLines(runInstruction);
+    for (TooLongLine tooLongLine : tooLongLines) {
+      ctx.reportIssue(TextRanges.range(tooLongLine.line, tooLongLine.startOffset, tooLongLine.line, tooLongLine.endOffset), MESSAGE);
+    }
+  }
+
+  private List<TooLongLine> computeTooLongLines(RunInstruction runInstruction) {
+    List<TooLongLine> result = new ArrayList<>();
+    var runInstructionData = computeRunInstructionDataPerLines(runInstruction);
+
+    for (Map.Entry<Integer, Integer> linesWithOffsets : runInstructionData.tooLongLinesWithLastOffset.entrySet()) {
+      int line = linesWithOffsets.getKey();
+      int startOffset = runInstructionData.firstOffsetPerLine.get(line);
+      int endOffset = linesWithOffsets.getValue();
+      if (runInstructionData.wordsPerLine.getOrDefault(line, 0) >= minWordsToTrigger) {
+        result.add(new TooLongLine(line, startOffset, endOffset));
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Gather data about a {@link RunInstruction}:
+   * - the first offset of each line, required to raise issue precisely
+   * - the too long lines number with the last offset exceeding the {@link #maxLength} value
+   * - the number of words per lines, to not raise an issue when below the {@link #minWordsToTrigger} value
+   */
+  private RunInstructionData computeRunInstructionDataPerLines(RunInstruction runInstruction) {
+    Map<Integer, Integer> wordsPerLine = countWordsPerLine(runInstruction);
+    var runInstructionData = new RunInstructionData(new HashMap<>(), new HashMap<>(), wordsPerLine);
+
+    TreeVisitor<TreeContext> visitor = new TreeVisitor<>();
+    visitor.register(SyntaxToken.class, (TreeContext ctx, SyntaxToken token) -> {
+      runInstructionData.firstOffsetPerLine.computeIfAbsent(token.textRange().start().line(), line -> token.textRange().start().lineOffset());
+      // Special case of token that finish on a different line: consider the first offset of the end line to start at 0.
+      if (token.textRange().start().line() != token.textRange().end().line()) {
```

Comment:
We only have access to start and end line, so if there is a syntax token on 3 lines with a too long line in between we won't detect it? Maybe something to represent in tests

Reply:
Yes you are correct.
I will add a test to this.